server/src/gnome/prompter.rs

```diff
@@ -0,0 +1,287 @@
+#[zvariant(signature = "a{sv}")]
+// System prompt properties
+pub struct Properties {
+    title: Option<String>,
+    #[zvariant(rename = "choice-label")]
+    choice_label: Option<String>,
+    description: Option<String>,
+    message: Option<String>,
+    #[zvariant(rename = "caller-window")]
+    caller_window: Option<String>,
+    warning: Option<String>,
+    #[zvariant(rename = "password-new")]
+    password_new: Option<bool>,
+    #[zvariant(rename = "password-strength")]
+    password_strength: Option<u32>,
+    #[zvariant(rename = "choice-chosen")]
+    choice_chosen: Option<bool>,
+    #[zvariant(rename = "continue-label")]
+    continue_label: Option<String>,
+    #[zvariant(rename = "cancel-label")]
+    cancel_label: Option<String>,
+}
+
+#[derive(Debug, Type)]
+#[zvariant(signature = "s")]
+// Possible values for PromptReady reply parameter
+pub enum Reply {
+    No,
+    Yes,
+    Empty,
+}
+
+const NO: &str = "no";
+const YES: &str = "yes";
+const EMPTY: &str = "";
+
+impl Serialize for Reply {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            Self::No => str::serialize(NO, serializer),
+            Self::Yes => str::serialize(YES, serializer),
+            Self::Empty => str::serialize(EMPTY, serializer),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for Reply {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        match String::deserialize(deserializer)?.as_str() {
+            NO => Ok(Self::No),
+            YES => Ok(Self::Yes),
+            EMPTY => Ok(Self::Empty),
+            err => Err(serde::de::Error::custom(format!("Invalid reply {err}"))),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Type)]
+#[serde(rename_all = "lowercase")]
+#[zvariant(signature = "s")]
+// Possible values for PerformPrompt type parameter
+pub enum PromptType {
+    Confirm,
+    Password,
+}
+
+// org.gnome.keyring.internal.Prompter
+
+#[proxy(
+    default_service = "org.gnome.keyring.SystemPrompter",
+    interface = "org.gnome.keyring.internal.Prompter",
+    default_path = "/org/gnome/keyring/Prompter"
+)]
+pub trait Prompter {
+    fn begin_prompting(&self, callback: &OwnedObjectPath) -> Result<(), ServiceError>;
+
+    fn perform_prompt(
+        &self,
+        callback: OwnedObjectPath,
+        type_: PromptType,
+        properties: Properties,
+        exchange: &str,
+    ) -> Result<(), ServiceError>;
+
+    fn stop_prompting(&self, callback: OwnedObjectPath) -> Result<(), ServiceError>;
+}
+
+// org.gnome.keyring.internal.Prompter.Callback
+
+pub struct PrompterCallback {
+    service: Service,
+    path: OwnedObjectPath,
+}
+
+#[interface(name = "org.gnome.keyring.internal.Prompter.Callback")]
+impl PrompterCallback {
+    pub async fn prompt_ready(
+        &self,
+        reply: Reply,
+        _properties: Properties,
```

Comment:
Why is this unused? Already mentioned it before but you marked the thread as resolved except not really...

Reply:
I'll take another look.